### PR TITLE
chore: update a/b testing docs on eigen

### PR DESCRIPTION
**Pull request by MounirDhahri**
This PR resolves [ONYX-808] 

### Description

This PR is an effort to make it easier to set up successful A/b experiments in Eigen in the future. It comes as a follow-up on this [diagnosis](https://www.notion.so/artsy/A-B-Testing-in-Eigen-7b3d8804f46f4f11a81c6bbd53504c1a?p=d4e8307871844fb3be2eb05d519adece&pm=s) and these additions (https://github.com/artsy/eigen/pull/9959 and https://github.com/artsy/eigen/pull/9960 and https://github.com/artsy/eigen/pull/9963 and https://github.com/artsy/eigen/pull/9965)
-  Added A/b testing best practices doc
- Update `add_an_experiment` to ~**not rely on the variant**~ and instead **rely on the payload** for rendering the UI
- Add docs on how to add an override

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- update a/b testing docs - mounir

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[ONYX-808]: https://artsyproduct.atlassian.net/browse/ONYX-808?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Dev changes (update a/b testing docs - mounir)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against fe48a7f7df9b10b711ecfc073e9c4574e07350fa
</p>

File: docs/a_b_testing_best_practices.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+> It's important to ship experiments with the least updates and bug fixes to them as possible. This is important because it **can** have an impact on the experiment results.
+
+Below is a non exhaustive list of areas to QA
+
+- [ ] The control variant functionality.
+- [ ] The experiment variant functionality.
+- [ ] Tracking for both the control and experiment variants.
+  - [ ] The tracking event is triggered only once per screen!
+  - [ ] This is important to make sure that we don't end up triggering the same event multiple times and breaking the split ratio.
```
Comment (damassi) [inline docs/a_b_testing_best_practices.md:25]: 👍

File: docs/a_b_testing_best_practices.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+Want to learn how to switch between different variants, see how to [add an override](add_an_experiment.md#adding-an-override)
+
+## Ready for release checkboxes!
+
+- [ ] My feature has been QA'd and it was approved by product, design, and data!
+  - [ ] Future updates or bug fixes,
```
Comment (damassi) [inline docs/a_b_testing_best_practices.md:32]: This line is a bit unclear. Can we expand it a bit with more context?

File: docs/a_b_testing_best_practices.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+- [ ] My feature related feature flag `readyForRelease` is set to `true`.
+- [ ] My feature feature flag is available in Echo and is set to `true`
```
Comment (damassi) [inline docs/a_b_testing_best_practices.md:34]: Maybe "feature related feature flag", similar to the bullet point above?

File: docs/a_b_testing_best_practices.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+- [ ] My feature experiment flag is enabled for all users in `unleash.artsy.net`
```
Comment (damassi) [inline docs/a_b_testing_best_practices.md:35]:
```suggestion
- [ ] My feature **experiment** flag is enabled for all users in `unleash.artsy.net`
```

Just to distinguish that this is a different type of flag, at peripheral glance

File: docs/a_b_testing_best_practices.md (hunk 2 above)
Comment (MounirDhahri) [reply docs/a_b_testing_best_practices.md:32]: this is not supposed to be here, I removed this line (or at least I thought so)

File: docs/a_b_testing_best_practices.md (hunk 4 above)
Comment (MounirDhahri) [reply docs/a_b_testing_best_practices.md:35]: good idea! will do it

File: docs/a_b_testing_best_practices.md (hunk 3 above)
Comment (MounirDhahri) [reply docs/a_b_testing_best_practices.md:34]: 🙏

File: docs/add_an_experiment.md [hunk 5]
```diff
@@ -58,12 +75,14 @@ You can access the variant value in a functional react component using `useExper
...
     <>
-+    {ourNewExperiment.enabled && ourNewExperiment.variant === "varA" && <AComponent />}
-+    {ourNewExperiment.enabled && ourNewExperiment.variant === "varB" && <BComponent custom={ourNewExperiment.payload} />}
++    {ourNewExperiment.enabled && ourNewExperiment.payload === "payloadA" && <AComponent />}
++    {ourNewExperiment.enabled && ourNewExperiment.payload === "payloadB" && <BComponent custom={ourNewExperiment.payload} />}
     <>
   )
 ```
 
+> Note: Avoid using `experiment.variant` and instead use the `experiment.payload` for rendering your UI! it's more future proof and it's more convenient this way to create multiple experiments using the same flag where you update the control variant
```
Comment (damassi) [inline docs/add_an_experiment.md:84]: Not for this PR, but maybe a lint rule or a Danger rule that looks for `.variant` would be good? I can imagine this being easily missed (as "variant", when working with variants, feels natural from an API perspective). 

No actions needed here, but lets keep an eye on things.

File: docs/add_an_experiment.md [hunk 6]
```diff
@@ -80,6 +99,20 @@ For example, in the previous experiment, we were testing if `varA` is performing
 Then we need to update eigen to remove the experiment code and use only the winner variant code.
 After that, we can archive that experiment in the Unleash dashboard.
 
+## Adding an Override
+
+Our infra supports adding admin overrides from the dev menu. In order to add an override:
+
+1. Open the dev menu
+2. Tap on **Experiments**
+3. Tap on ✏️ next to the payload or the variant name.
```
Comment (damassi) [inline docs/add_an_experiment.md:108]: Ha, that lil character magically turns into ✏️

**Review by damassi — DISMISSED** [review]
Excellent work 👍 Just a couple of minor suggestions

File: docs/a_b_testing_best_practices.md [hunk 7]
```diff
@@ -0,0 +1,36 @@
+# A/b Testing best practices
+
+## What will you need to run a successful experiment?
+
+1. An admin feature flag while you are developing the feature.
+   > This flag has only one goal: **Is the feature ready to be released to all users?**
```
Comment (joeyAghion) [inline docs/a_b_testing_best_practices.md:6]: Is this separate flag necessary, even with the developer tooling you added to override variants?

**Review by joeyAghion — APPROVED** [review]

File: docs/add_an_experiment.md (hunk 5 above)
Comment (MounirDhahri) [reply docs/add_an_experiment.md:84]: Nice idea, will add that tomorrow ;)

File: docs/a_b_testing_best_practices.md (hunk 7 above)
Comment (MounirDhahri) [reply docs/a_b_testing_best_practices.md:6]: Unfortunately, currently yes, it's required. The reason behind that is that we have no way yet to release a feature from a specific build with our unleash infra. This rings a bell for me though, i will follow up with that separately

File: docs/a_b_testing_best_practices.md [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+  - [ ] The tracking event is triggered only once per screen!
+  - [ ] This is important to make sure that we don't end up triggering the same event multiple times and breaking the split ratio.
+
+Want to learn how to switch between different variants, see how to [add an override](add_an_experiment.md#adding-an-override)
+
+## Ready for release checkboxes!
+
+- [ ] My feature has been QA'd and it was approved by product, design, and data!
+- [ ] My feature related feature flag `readyForRelease` is set to `true`.
+- [ ] My feature related feature flag is available in Echo and is set to `true`
```
Comment (joeyAghion) [inline docs/a_b_testing_best_practices.md:33]: "Feature related feature flag" is... odd. What is that meant to distinguish? Maybe we can use "toggles" and "experiments" instead.

Comment (MounirDhahri) [reply docs/a_b_testing_best_practices.md:33]: toggle is good! will change

**Comment by MounirDhahri** [comment]
Addressed your comments and will merge this. Glad to address further comments separately

File: docs/a_b_testing_best_practices.md (hunk 7 above)
Comment (MounirDhahri) [reply docs/a_b_testing_best_practices.md:6]: Here it is https://github.com/artsy/README/issues/536

**Pull request merged**